src/psyclone/tests/psyclone_test_utils.py

```diff
@@ -340,17 +326,12 @@ def string_compiles(self, code):
         with open(filename, 'w') as test_file:
             test_file.write(code)
 
+        success = True
         try:
-            success = self.compile_file(filename)
+            self.compile_file(filename)
```

Comment:
Now that you've simplified this, you could simplify it further by just doing `return False` in the `except CompileError` part and `return True` at the end of the routine.

Reply:
(and pylint complains that old_pwd is not used)

Reply:
Actually ... the warning correctly indicates that I am not changing back to the original directory. While I don't think this will break anything (afaik any code using a temporary directory will chdir first anyway), but as long as we don't have this as a requirement, I think it's better that I add a finally and chdir back to tmpdir - which then also means keeping the success variable around.

Reply:
OK, I'm happy with that - it seems the safest thing to do.